### PR TITLE
converted ItMonitorintExporter tests to junit5

### PR DESCRIPTION
**Pull request by marinakog**
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-model-in-image-tests-10/416
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/502/

### CODE REVIEW

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMonitoringExporter.java [hunk 1]
```diff
@@ -226,112 +256,380 @@ public static void initAll(@Namespaces(6) List<String> namespaces) {
...
 
     // create and verify WebLogic domain image using model in image with model files
-    String imageName = createAndVerifyDomainImage();
+    miiImage = createAndVerifyMiiImage();
 
-    // create and verify one cluster domain
+    // create and verify one cluster mii domain
     logger.info("Create domain and verify that it's running");
-    createAndVerifyDomain(imageName, domain1Uid);
+    createAndVerifyDomain(miiImage, domain1Uid, domain1Namespace, "FromModel", 1);
```
Comment (sankarpn) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMonitoringExporter.java:263]: Dongbo has refactored a method to use common code, please see if that is useful here.
https://github.com/oracle/weblogic-kubernetes-operator/pull/1759/files#diff-563c7e0923febf5308b79abec0486d46

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMonitoringExporter.java [hunk 2]
```diff
@@ -226,112 +256,380 @@ public static void initAll(@Namespaces(6) List<String> namespaces) {
...
   /**
-   * Test covers the following use cases.
+   * Test covers end to end sample, provided in the Monitoring Exporter github project .
    * Create Prometheus, Grafana, Webhook, Coordinator
-   * create domain Model in Image with monitoring exporter
+   * create domain in Image with monitoring exporter
    * verify access to monitoring exporter WebLogic metrics via nginx
-   * check generated by monitoring exporter WebLogic metrics via Prometheus
+   * check generated by monitoring exporter WebLogic metrics via Prometheus, Grafana
+   * fire Alert using Webhook
```
Comment (sankarpn) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMonitoringExporter.java:297]: Please fix punctuation. It looks like this when javadoc is viewed.

Test covers end to end sample, provided in the Monitoring Exporter github project . Create Prometheus, Grafana, Webhook, Coordinator create domain in Image with monitoring exporter verify access to monitoring exporter WebLogic metrics via nginx check generated by monitoring exporter WebLogic metrics via Prometheus, Grafana fire Alert using Webhook change prometheus to add different domain to monitor

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMonitoringExporter.java [hunk 3]
```diff
@@ -226,112 +256,380 @@ public static void initAll(@Namespaces(6) List<String> namespaces) {
...
+
+      //verify access to Monitoring Exporter
+      verifyMonExpAppAccessThroughNginx(ingressHost2List.get(0), managedServersCount);
+      //verify metrics via prometheus
+      String testappPrometheusSearchKey =
+              "wls_servlet_invocation_total_count%7Bapp%3D%22test-webapp%22%7D%5B15s%5D";
+      checkMetricsViaPrometheus(testappPrometheusSearchKey, "test-webapp");
+      //fire alert by scaling down
+      fireAlert();
+      //switch to monitor another domain
```
Comment (sankarpn) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMonitoringExporter.java:326]: The above comments can be logged as messages, easier to debug when test fails.

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMonitoringExporter.java [hunk 4]
```diff
@@ -226,112 +256,380 @@ public static void initAll(@Namespaces(6) List<String> namespaces) {
...
-    installPrometheusGrafana();
-    installWebhookCoordinator();
+  /**
+   * Test covers end to end sample, provided in the Monitoring Exporter github project .
+   * Create Prometheus, Grafana
+   * create Model in Image with monitoring exporter
+   * verify access to monitoring exporter WebLogic metrics via nginx
+   * check generated by monitoring exporter WebLogic metrics via Prometheus, Grafana
+   * check basic functionality of monitoring exporter
+   */
```
Comment (sankarpn) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMonitoringExporter.java:349]: Fix javadoc

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMonitoringExporter.java [hunk 5]
```diff
@@ -602,65 +921,70 @@ private static void createCoordinator(String image,
...
+                                                      .name(secretName))))));
+
+      logger.info("Create deployment for coordinator in namespace {0}",
+              namespace);
+      boolean deploymentCreated = assertDoesNotThrow(() -> Kubernetes.createDeployment(coordinatorDepl),
+              String.format("Create deployment failed with ApiException for coordinator in namespace %s",
+                      namespace));
+      assertTrue(deploymentCreated, String.format(
+              "Create deployment failed with ApiException for coordinator in namespace %s ",
+              namespace));
```
Comment (sankarpn) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMonitoringExporter.java:963]: You might want to wait until the job is done and report any error if it fails. Please see createDomainJob in ItIntrospectVersion

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMonitoringExporter.java [hunk 6]
```diff
@@ -816,96 +1147,177 @@ private static String createAndVerifyDomainImage() {
...
-
+        domainUid, namespace, miiImage);
+    createDomainCrAndVerify(adminSecretName, REPO_SECRET_NAME, encryptionSecretName, miiImage,domainUid,
+            namespace, domainHomeSource, replicaCount);
+    String adminServerPodName = domainUid + "-admin-server";
     // check that admin server pod exists in the domain namespace
     logger.info("Checking that admin server pod {0} exists in namespace {1}",
-        adminServerPodName, domain1Namespace);
-    checkPodExists(adminServerPodName, domainUid, domain1Namespace);
+        adminServerPodName, namespace);
```
Comment (sankarpn) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMonitoringExporter.java:1262]: you don't need this check, podReady does both

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMonitoringExporter.java [hunk 7]
```diff
@@ -816,96 +1147,177 @@ private static String createAndVerifyDomainImage() {
...
-    checkPodReady(adminServerPodName, domainUid, domain1Namespace);
+        adminServerPodName, namespace);
+    checkPodReady(adminServerPodName, domainUid, namespace);
 
     // check that admin service exists in the domain namespace
     logger.info("Checking that admin service {0} exists in namespace {1}",
-        adminServerPodName, domain1Namespace);
-    checkServiceExists(adminServerPodName, domain1Namespace);
+        adminServerPodName, namespace);
+    checkServiceExists(adminServerPodName, namespace);
```
Comment (sankarpn) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMonitoringExporter.java:1273]: this has to be done before podReady, pod will never become ready when service creation fails.

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Kubernetes.java [hunk 8]
```diff
@@ -1117,6 +1118,50 @@ public static boolean createConfigMap(V1ConfigMap configMap) throws ApiException
     return true;
   }
 
+  /**
+   * Create a Kubernetes Config Map.
```
Comment (sankarpn) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Kubernetes.java:1122]:
```suggestion
   * Updates a Kubernetes Config Map.
```

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMonitoringExporter.java [hunk 9]
```diff
@@ -226,112 +256,380 @@ public static void initAll(@Namespaces(6) List<String> namespaces) {
...
+    }
   }
 
   /**
-   * Install Prometheus, Grafana using helm chart, and verify that pods are running.
-   * @throws ApiException when creating helm charts or pods fails
+   * Test covers the following use cases.
+   * Create Prometheus, Grafana from latest version of helm chart
+   * verify access to monitoring exporter WebLogic metrics via nginx
+   * check generated by monitoring exporter WebLogic metrics via Prometheus
```
Comment (bhavaniravichandran) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMonitoringExporter.java:410]: remove the "by"

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMonitoringExporter.java (hunk 1 above)
Comment (marinakog) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMonitoringExporter.java:263]: I add request to add domain creation based on custom image name, right now it uses prebuilt image. in my test I have prop files based image creation

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMonitoringExporter.java (hunk 2 above)
Comment (marinakog) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMonitoringExporter.java]: done

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMonitoringExporter.java (hunk 3 above)
Comment (marinakog) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMonitoringExporter.java]: done

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMonitoringExporter.java (hunk 4 above)
Comment (marinakog) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMonitoringExporter.java:342]: done

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMonitoringExporter.java (hunk 5 above)
Comment (marinakog) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMonitoringExporter.java:967]: added check for depl

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMonitoringExporter.java (hunk 6 above)
Comment (marinakog) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMonitoringExporter.java]: done

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Kubernetes.java (hunk 8 above)
Comment (marinakog) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Kubernetes.java]: done

**Comment by marinakog** [comment]
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/517
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-model-in-image-tests-10/426/

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Kubernetes.java [hunk 10]
```diff
@@ -1117,6 +1118,50 @@ public static boolean createConfigMap(V1ConfigMap configMap) throws ApiException
     return true;
   }
 
+  /**
+   * Replace a Kubernetes Config Map.
```
Comment (anpanigr) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Kubernetes.java:1122]: Can u explain bit more detail. What is the purpose of replacing a Config Map

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/assertions/impl/Kubernetes.java [hunk 11]
```diff
@@ -429,6 +433,74 @@ public static V1Service getService(
...
+          }
+        }
+        return deployment;
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Checks if an deployment  is running in a given namespace.
```
Comment (anpanigr) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/assertions/impl/Kubernetes.java:476]: extra blank space between the word  "deployment"  and "is"

File: new-integration-tests/src/test/resources/exporter/exporter-config.yml [hunk 12]
```diff
@@ -0,0 +1,61 @@
+# Copyright (c) 2020, Oracle Corporation and/or its affiliates.
```
Comment (anpanigr) [inline new-integration-tests/src/test/resources/exporter/exporter-config.yml:1]: rename the file name to exporter-config.yaml to be consistent with yaml file extension with other similar files in resource directory.

File: new-integration-tests/src/test/resources/exporter/rest_domainqualtrue.yml [hunk 13]
```diff
@@ -0,0 +1,23 @@
+# Copyright (c) 2020, Oracle Corporation and/or its affiliates.
```
Comment (anpanigr) [inline new-integration-tests/src/test/resources/exporter/rest_domainqualtrue.yml:1]: Consider rename the file name to rest_domainqualtrue.yaml to be consistent with yaml file extension with other similar files in resource directory.

File: new-integration-tests/src/test/resources/exporter/rest_dublicatedval.yml [hunk 14]
```diff
@@ -0,0 +1,21 @@
+# Copyright (c) 2019, 2020, Oracle Corporation and/or its affiliates.
```
Comment (anpanigr) [inline new-integration-tests/src/test/resources/exporter/rest_dublicatedval.yml:1]: Same as other yaml files

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMonitoringExporter.java [hunk 15]
```diff
@@ -204,134 +231,398 @@ public static void initAll(@Namespaces(6) List<String> namespaces) {
...
+    }
+  }
+
+  /**
+   * Test covers end to end sample, provided in the Monitoring Exporter github project .
+   * Create Prometheus, Grafana
+   * create Model in Image with monitoring exporter
    * verify access to monitoring exporter WebLogic metrics via nginx
-   * check generated by monitoring exporter WebLogic metrics via Prometheus
+   * check generated by monitoring exporter WebLogic metrics via Prometheus, Grafana
```
Comment (bhavaniravichandran) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMonitoringExporter.java:340]: Remove "by" from the comment above

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMonitoringExporter.java [hunk 16]
```diff
@@ -204,134 +231,398 @@ public static void initAll(@Namespaces(6) List<String> namespaces) {
...
+    }
   }
 
   /**
-   * Install Prometheus, Grafana using helm chart, and verify that pods are running.
-   * @throws ApiException when creating helm charts or pods fails
+   * Test covers the following use cases.
+   * Create Prometheus, Grafana from latest version of helm chart
+   * verify access to monitoring exporter WebLogic metrics via nginx
+   * check WebLogic metrics via Prometheus
```
Comment (bhavaniravichandran) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMonitoringExporter.java:403]: Fix the javadoc. Similar to Sankar's comment earlier.

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMonitoringExporter.java [hunk 17]
```diff
@@ -204,134 +231,398 @@ public static void initAll(@Namespaces(6) List<String> namespaces) {
...
    */
-  private void installPrometheusGrafana() throws IOException, ApiException {
-    createPvAndPvc("prometheus");
-    createPvAndPvc("alertmanager");
-    createPvAndPvc("grafana");
+  private void checkPromGrafanaLatestVersion() throws Exception {
+    //check if helm was already created and uninstall
```
Comment (bhavaniravichandran) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMonitoringExporter.java:406]: fix the comment above - not uninstalling helm here are you?

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMonitoringExporter.java [hunk 18]
```diff
@@ -1018,8 +1438,404 @@ public static boolean execCommandCheckResponse(String cmd, String searchKey) {
...
+
+  private void changeConfigNegativeAuth(
+          String effect, String configFile, String expectedErrorMsg, String username, String password)
+          throws Exception {
+    try {
+      final WebClient webClient = new WebClient();
+      //webClient.addRequestHeader("Host", ingressHost1List.get(0));
+      //webClient.addRequestHeader("Location",exporterUrl);
+      //webClient.addRequestHeader("Referer", exporterUrl);
+
```
Comment (bhavaniravichandran) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMonitoringExporter.java:1493]: remove commented code above

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/CommonTestUtils.java [hunk 19]
```diff
@@ -803,19 +838,19 @@ public static String createMiiImageAndVerify(String miiImageNameBase,
...
 
     final List<String> wdtModelList = Collections.singletonList(MODEL_DIR + "/" + wdtModelFile);
     final List<String> appSrcDirList = Collections.singletonList(appName);
     final List<String> modelPropList = Collections.singletonList(altModelDir + "/" + modelPropFile);
 
     return createImageAndVerify(
-        imageNameBase, wdtModelList, appSrcDirList, modelPropList, WLS_BASE_IMAGE_NAME,
-        WLS_BASE_IMAGE_TAG, WLS, false, domainHome, false);
+            imageNameBase, wdtModelList, appSrcDirList, modelPropList, WLS_BASE_IMAGE_NAME,
+            WLS_BASE_IMAGE_TAG, WLS, false, domainHome, false);
```
Comment (bhavaniravichandran) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/CommonTestUtils.java:853]: Are there tabs? I see it in a lot of files actually.

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/CommonTestUtils.java [hunk 20]
```diff
@@ -942,43 +977,43 @@ public static String createImageAndVerify(String imageNameBase,
...
+                      .modelImageTag(imageTag)
+                      .modelFiles(wdtModelList)
+                      .modelVariableFiles(modelPropList)
+                      .modelArchiveFiles(archiveList)
+                      .domainHome(WDT_IMAGE_DOMAINHOME_BASE_DIR + "/" + domainHome)
+                      .wdtModelOnly(modelType)
+                      .wdtOperation("CREATE")
+                      .wdtVersion(WDT_VERSION)
+                      .env(env)
+                      .redirect(true));
```
Comment (bhavaniravichandran) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/CommonTestUtils.java:994]: Looks like tabs here again.

Comment (marinakog) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/CommonTestUtils.java]: done

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/CommonTestUtils.java (hunk 19 above)
Comment (marinakog) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/CommonTestUtils.java]: done

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMonitoringExporter.java (hunk 18 above)
Comment (marinakog) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMonitoringExporter.java]: done

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMonitoringExporter.java (hunk 17 above)
Comment (marinakog) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMonitoringExporter.java]: done

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMonitoringExporter.java (hunk 16 above)
Comment (marinakog) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMonitoringExporter.java:408]: done

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMonitoringExporter.java (hunk 15 above)
Comment (marinakog) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMonitoringExporter.java]: done

File: new-integration-tests/src/test/resources/exporter/rest_dublicatedval.yml (hunk 14 above)
Comment (marinakog) [reply new-integration-tests/src/test/resources/exporter/rest_dublicatedval.yml]: done

File: new-integration-tests/src/test/resources/exporter/rest_domainqualtrue.yml (hunk 13 above)
Comment (marinakog) [reply new-integration-tests/src/test/resources/exporter/rest_domainqualtrue.yml]: done

File: new-integration-tests/src/test/resources/exporter/exporter-config.yml (hunk 12 above)
Comment (marinakog) [reply new-integration-tests/src/test/resources/exporter/exporter-config.yml]: done

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/assertions/impl/Kubernetes.java (hunk 11 above)
Comment (marinakog) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/assertions/impl/Kubernetes.java]: done

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Kubernetes.java (hunk 10 above)
Comment (marinakog) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Kubernetes.java:1122]: this is to support updates for CM ( kubectl edit cm ) I have added more description

**Pull request merged**